Client/src/pages/suggestionHistory/suggestionHistory.tsx

```diff
@@ -5,97 +5,89 @@ import CalendarTodayIcon from '@mui/icons-material/CalendarToday';
 
-interface Suggestion {
-  title: string;
-  content: string;
-  rank: number;
-  date: Date;
-  eventTitle: string;
+interface FlatEvent {
+  eventId: number;
+  userId: number;
   eventContent: string;
-  eventMoodCategory: MoodCategory;
+  eventDate: string; 
+  mood: Mood;
+  suggestionId: number;
+  suggestionTitle: string;
+  suggestionContent: string;
+  suggestionRank: number; 
+  suggestionExecutionDate: string;
+  // keywords: [];
 }
 
-enum MoodCategory {
-  Positive = "Positive",
-  Negative = "Negative",
-  Regular = "Regular",
-}
-
-const suggestionsList: Suggestion[] = [
-  {
-    eventTitle: "Morning Jog",
-    eventContent: "Went for a 5km jog in the park. Felt great.",
-    title: "Stretching",
-    content: "Do some stretching exercises to cool down.",
-    rank: 2,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Positive
-  },
-  {
-    eventTitle: "Evening Walk",
-    eventContent: "Had a relaxing walk in the neighborhood.",
-    title: "Hydrate",
-    content: "Drink plenty of water after your walk.",
-    rank: 3,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Regular
-  },
-  {
-    eventTitle: "Evening Walk",
-    eventContent: "Had a relaxing walk in the neighborhood.",
-    title: "Hydrate",
-    content: "Drink plenty of water after your walk.",
-    rank: 1,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Negative
-  },
-  {
-    eventTitle: "Evening Walk",
-    eventContent: "Had a relaxing walk in the neighborhood.",
-    title: "Hydrate",
-    content: "Drink plenty of water after your walk.",
-    rank: 3,
-    date: new Date(),
-    eventMoodCategory: MoodCategory.Regular
-  },
-  // Add more suggestions as needed
-];
-
 const HistoryOfSuggestionsPage: React.FC = () => {
-  const [suggestions, setSuggestios] = useState<Suggestion[]>([]);
+  const [suggestions, setSuggestios] = useState<FlatEvent[]>([]);
 
   useEffect(() => {
-    // Todo: get dailySharings of today
-    // const events: Suggestion[] = [];
+      async function fetchMyAPI() {
+        // TODO: Update to SERVER_URL
+        const response = await fetch('http://localhost:3000/events/?withSuggestions=true', {
+          method: 'GET',
+          headers: {
+            // TODO: Update to user token
+            'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MCwiaWF0IjoxNzIzNDg0NTc2LCJleHAiOjE3MjM1ODQ1NzZ9.1L_d4V7cGS7izOSuRlBi-6jO9JGGitkh44nALyePIDs',
+            'Content-Type': 'application/json',
+          },
+        });        
+
+        const result = await response.json();
+
+        const flatEvents : FlatEvent[] = result.flatMap((event: any)  =>
+          event.suggestions.map((suggestion: any) => ({
+              eventId: event.id,
+              userId: event.user_id,
+              eventContent: event.content,
+              eventDate: event.date,
+              mood: event.mood,
+              suggestionId: suggestion.id,
```

Comment:
add all fields that are connected to suggestion to be inside suggestion field, like:
suggestion: {
id: ....
title: ....
etc
}

Reply:
why?